### PR TITLE
Add banner component

### LINKED ISSUE #385: Show or hide a banner based on allsearch_api response
### User story
As a library administrator, I sometimes want to prominently display temporary messages to users. 

### Acceptance criteria
- [ ] Uses the data from the allsearch-api for:
   - [ ] the banner text (default none)
   - [ ] the alert status (valid options: [warning, error, success, info]) (default: info)
   - [ ] whether or not the alert should be dismissible (default: true)
   - [ ] whether or not the alert should autoclear (default: false)
- [ ] Uses the data from the allsearch-api to show or hide the component (default: hide)
- [ ] Displays html from the allsearch-api response, including headers and links
- [ ] Use the Lux alert component
- [ ] Check for link display contrast (must be accessible)
- [ ] Check for display in dark mode (should be the same display in light or dark mode)
- [ ] I can successfully use this feature with Voice Over in Safari.
- [ ] I have checked this feature using one of the following browser extensions and confirmed that it does not add any level A or level AA issues:
    * axe dev tools
    * Lighthouse
    * Wave

### Concrete example
During the soft launch of the website, we want to solicit feedback from users, and to do so using a banner that includes a header and a link to a survey.

### Implementation notes, if any
See https://github.com/pulibrary/allsearch_api/issues/265 for allsearch-api implementation.

### PR DESCRIPTION
**Pull request by rladdusaw**
Resolves #385

### CODE REVIEW

**Review by maxkadel — CHANGES_REQUESTED** [review]
It looks as though the html is rendering as text, rather than as html. 

image

In order to make it safe to pass on as JSON, we're escaping the html in the api, but then we'd discussed un-escaping it for display.

**Review by maxkadel — DISMISSED** [review]
Much better! It's displaying the html now.

Two requests:
- Can we add it to the home page as well? (`/`)
- Is there some way to not have it move the page around mid-loading? Fo me it's rendering a tiny bit after the rest of the page, so the whole page moves.

**Comment by rladdusaw** [comment]
> * Is there some way to not have it move the page around mid-loading? Fo me it's rendering a tiny bit after the rest of the page, so the whole page moves.

I don't think there's any way to avoid that. It has to wait for the response from the API to decide if it's going to be displayed or not. We would have to reserve space for it and leave it empty if there was no alert. I'm open to alternative options, but I'm not thinking of anything better.

File: src/services/BannerService.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      baseURL: config.allsearchApiUrl
+    };
+    this.client = axios.create(options);
+  }
+
+  public result(): Promise<Banner> {
+    return this.client
+      .get<Banner>('/banner')
+      .then(res => plainToInstance(BannerResult, res.data))
+      .catch(() => new BannerResult('', false, 'info', true, false));
```
Comment (sandbergja) [inline src/services/BannerService.ts:23]: Nice, I like this error catching.

**Review by sandbergja — COMMENTED** [review]
Thanks, @rladdusaw !  It's looking pretty good to me.

- I agree about the layout shift; reserving some space sounds like a good solution to me.  Just fiddling with it in lighthouse, it looked like reserving 30px of vertical space (just wrapping it in a div with min-height) brought the Cumulative Layout Shift from .302 to .262 in the search results page, and it still looks fine to me even when there is no banner.  It still shifts for sure, but it's not as jarring.
- On the landing page, it still shifts even though there is a lot of whitespace above the search bar.
- This is probably future work, but if you dismiss the banner and try another search, the banner shows up again.  As a user, I probably would expect that when I dismiss the banner, it stays dismissed (maybe we could store that in localStorage or something?)

File: src/services/BannerService.ts (hunk 1 above)
Comment (rladdusaw) [reply src/services/BannerService.ts:23]: Thanks, I copied it from the SearchService 🤣

**Review by sandbergja — APPROVED** [review]
Thanks, @rladdusaw 🐈‍⬛

**Pull request merged**